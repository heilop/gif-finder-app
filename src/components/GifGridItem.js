import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({id, title, url}) => {

  return (
    <div className = 'card animate_animated animate_fadeIn'>
      <img src={ url } alt={ title }></img>
      <p>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
