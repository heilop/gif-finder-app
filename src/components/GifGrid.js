import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category }) => {

  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{ category }</h3>
      { loading && <p className="card-grid animate__animated animate__flash">Loading</p> }

      <div className="card-grid animate__animated animate__fadeIn">

        {
          images.map(img => (
            <GifGridItem
              key = { img.id }
              img = {img}
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
