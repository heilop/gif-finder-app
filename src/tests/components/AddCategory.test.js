import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Test in <AddCategory />', () => {

  const setCategories = () => {};
  test('should show the right component', () => {

    const wrapper = shallow(<AddCategory setCategories = { setCategories }/>);

    expect(wrapper).toMatchSnapshot();
  })

})
