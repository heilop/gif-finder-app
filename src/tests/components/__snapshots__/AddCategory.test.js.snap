// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AddCategory /> should show the right component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
